### DISCUSSION TITLE
Flux bootstrap git hash

### DISCUSSION THREAD

**Discussion by jack-evans** [fluxcd/flux2#4262] [Q&A]
Is it possible to run flux bootstrap with a commit hash?

Atm we have a branch based pipeline:

Dev -> master -> integration -> stage -> prod

If we moved to a mono-repo where everything is on master branch, I want to keep my changes following this format in the same automated fashion. How can I achieve this if I am always flux bootstrap master?

**Comment by stefanprodan** [comment]
Flux bootstrap is not made for app repos. You run bootstrap on a new repo dedicated to Flux, in there you create GitRepository or OCIRepository objects to point to app manifests using Git SHAs or Git tags or OCI digests or OCI tags to drive the app delivery.
